system_metrics_collector/src/system_metrics_collector/utilities.cpp

```diff
@@ -101,6 +136,27 @@ double computeCpuActivePercentage(
   return 100.0 * active_time / total_time;
 }
 
+double computePidCpuActivePercentage(
+  const ProcPidCpuData & process_measurement1,
+  const ProcCpuData & system_measurement1,
+  const ProcPidCpuData & process_measurement2,
+  const ProcCpuData & system_measurement2)
+{
+  if (process_measurement1.isMeasurementEmpty() || system_measurement1.isMeasurementEmpty() ||
+    process_measurement2.isMeasurementEmpty() || system_measurement2.isMeasurementEmpty())
+  {
+    RCUTILS_LOG_ERROR_NAMED("computePidCpuActivePercentage",
+      "a measurement was empty, unable to compute cpu percentage");
+    return std::nan("");
+  }
+
+  const double active_time = process_measurement2.getActiveTime() -
+    process_measurement1.getActiveTime();
```

Comment:
Does this handle the case where t1 > t2 because clock overflowed? See man clock:

> Note that the time can wrap around.  On a 32-bit system where CLOCKS_PER_SEC equals 1000000 this function will return the same value approximately every 72 minutes.